src/lib/input/input-container.ts

```diff
@@ -234,10 +249,20 @@ export class MdInputDirective {
   selector: 'md-input-container, mat-input-container',
   templateUrl: 'input-container.html',
   styleUrls: ['input-container.css'],
+  animations: [
+    trigger('transitionMessages', [
+      state('enter', style({ opacity: 1, transform: 'translateY(0%)' })),
+      transition('void => enter', [
+        style({ opacity: 0, transform: 'translateY(-100%)' }),
+        animate('300ms')
```

Comment:
Can you add in an easing function to avoid a harsh linear translation?
https://material.io/guidelines/motion/duration-easing.html#duration-easing-natural-easing-curves